push_notifications/gcm.py

```diff
@@ -120,7 +120,7 @@ def _gcm_send_json(registration_ids, data, **kwargs):
 	data = json.dumps(values, separators=(",", ":"), sort_keys=True).encode("utf-8")
 
 	response = json.loads(_gcm_send(data, "application/json"))
-	if response["failure"] or response["canonical_ids"]:
+	if response.get("failure", None) or response.get("canonical_ids", None):
```

Comment:
The second argument is not necessary, `.get()` will return `None` if you do not give it a second argument.